autopush_rs/src/util/megaphone.rs

```diff
@@ -198,6 +224,31 @@ impl ServiceChangeTracker {
             svc_delta,
         )
     }
+
+    /// Update a `ClientServices` to account for a new service.
+    ///
+    /// Returns services that have changed.
+    pub fn client_service_add_service(&self, client_service: &mut ClientServices, services: &[Service]) -> Option<Vec<Service>> {
+        let mut svc_delta = self.change_count_delta(client_service).unwrap_or(Vec::new());
+        for svc in services.iter() {
```

Comment:
off hand i don't know if it's easily refactorable/doable but: service_delta has a very similar loop here

Reply:
Indeed, they both need to do slightly different things though, so I didn't see any way to abstract it out.